misc/multiaddr/src/lib.rs

```diff
@@ -52,6 +53,11 @@ impl Multiaddr {
         Vec::from(&self.bytes[..])
     }
 
+    /// Builds a [`Multiaddr`] from a byte representation.
+    pub fn from_vec(vec: Vec<u8>) -> Self {
```

Comment:
This allows creating invalid multi-addresses from arbitrary byte vectors, right? I don't think that is desirable. I think any such conversion should go through one of the existing `TryFrom` instances.